src/comm/AcceptLimiter.cc

```diff
@@ -51,7 +51,9 @@ Comm::AcceptLimiter::kick()
         deferred_.erase(deferred_.begin());
         if (temp.valid()) {
             debugs(5, 5, "doing one.");
-            temp->acceptNext();
+            typedef CommCbMemFunT<Comm::TcpAcceptor, CommIoCbParams> Dialer;
+            AsyncCall::Pointer call = JobCallback(33, 5, Dialer, temp.get(), TcpAcceptor::acceptOne);
+            ScheduleCallHere(call);
             break;
```

Comment:
As detailed at https://github.com/squid-cache/squid/pull/857#issuecomment-875976489 , quitting this kick() loop without ensuring progress stalls queued TcpAcceptors. To better understand the problem, treat scheduling of basic async calls to other jobs as a no-op (because that scheduling will result in no relevant state changes -- no progress -- in some cases).

Reply:
This is fixed by ad0a9e6. We are now hitting consequences of Comm::Read() when the OS closes the listening socket at shutdown.